### PR TITLE
Append library node expansions to an SDFG's history

### PR DESCRIPTION
**Pull request by phschaad**

### CODE REVIEW

**Review by tbennun — CHANGES_REQUESTED** [review]
I'd say this should happen after `apply` is complete.

**Comment by phschaad** [comment]
That's a good point. Should I make the same change to all transformation applications? At the moment, `apply_pattern` appends before applying too.

**Comment by tbennun** [comment]
@phschaad sure

File: dace/transformation/transformation.py [hunk 1]
```diff
@@ -535,6 +537,39 @@ def apply(self, sdfg, *args, **kwargs):
...
+            'type': 'ExpandTransformation',
+            'transformation': type(self).__name__,
+            'classpath': nd.full_class_path(self),
+            **props
+        }
+
+    @staticmethod
+    def from_json(json_obj: Dict[str, Any],
+                  context: Dict[str, Any] = None) -> 'ExpandTransformation':
+        xform = pydoc.locate(json_obj['classpath'])
```
Comment (tbennun) [inline dace/transformation/transformation.py:552]: what if it doesn't exist? does it become a serializableobject?

Comment (phschaad) [reply dace/transformation/transformation.py:552]: Yes, exactly - it diverts to `Transformation.from_json`, which in turn fails and generates a SerializableObject

**Review by tbennun — APPROVED** [review]

**Pull request merged**